sa/model.go

```diff
@@ -88,14 +88,18 @@ func ClearEmail(ctx context.Context, dbMap db.DatabaseMap, regID int64, email st
 			return nil, nil
 		}
 
-		currPb.Contact = newContacts
-		newModel, err := registrationPbToModel(currPb)
-		if err != nil {
-			return nil, err
-		}
+		// UPDATE the row with a direct database query, in order to avoid LockCol issues.
+		//
+		// TODO(#7716): Revert to use tx.Update() once LockCol has been removed.
+		_, err = dbMap.ExecContext(ctx,
```

Comment:
I think this needs to be `tx.ExecContext`, so that it happens as part of the same transaction as the `selectRegistration` on line 69.

For context: the `dbMap` is passed into the `ClearEmail` function, and then immediately passed through into the `WithTransaction` function as its second argument. That function uses it to create a `db.Executor` object and passes _that_ in to the anonymous `func (tx db.Executor) (interface{}, error)` function that is `WithTransaction`'s third argument. We're inside the body of that anonymous function right now (it ends on line 101), so we need to be using the `tx` object instead of the original `dbMap` object.